src/lib/migration-chunks/validation/editor-layout.ts

```diff
@@ -0,0 +1,163 @@
+import Intent from '../../interfaces/intent'
+import { EditorInterfaces } from '../../entities/content-type'
+import { InvalidActionError } from '../../interfaces/errors'
+import errors from './errors'
+import { collectFieldGroupIds } from '../../utils/editor-layout'
+const editorLayoutErrors = errors.editorLayout
+
+interface ValidationContext {
+  remoteFieldGroups: Set<string>
+  createdFieldGroups: Set<string>
+  deletedFieldGroups: Set<string>
+  toBeCreatedFieldGroups: Set<string>
+}
+
+interface EditorLayoutValidation {
+  validate (intent: Intent, context: ValidationContext): string | string[]
+}
+
+class DuplicateCreate implements EditorLayoutValidation {
+  validate (intent: Intent, context: ValidationContext) {
+    if (!intent.isFieldGroupCreate()) {
+      return
```

Comment:
Shouldn't those guard clauses return empty arrays according to the interface above?

Reply:
Idk why ts does not complain (maybe because our return type is a super set of the interface?) but `undefined` as possible return looks better for me. Anyway yes, an empty string or array would behave the same.